src/main/java/io/vertx/mqtt/MqttTopicSubscription.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2016 Red Hat Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.vertx.mqtt;
+
+import io.netty.handler.codec.mqtt.MqttQoS;
+import io.vertx.codegen.annotations.VertxGen;
+
+/**
+ * Represents a subscription to a topic
+ */
+@VertxGen
+public interface MqttTopicSubscription {
+
+  /**
+   * Subscription topic name
+   *
+   * @return
+   */
+  String topicName();
```

Comment:
@vietj can you explain the main role of CacheReturn and why it should be used here ? Thanks !